python/ql/lib/semmle/python/frameworks/RestFramework.qll

```diff
@@ -0,0 +1,302 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `djangorestframework` PyPI package
+ * (imported as `rest_framework`)
+ *
+ * See
+ * - https://www.django-rest-framework.org/
+ * - https://pypi.org/project/djangorestframework/
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+private import semmle.python.frameworks.internal.InstanceTaintStepsHelper
+private import semmle.python.frameworks.Django
+private import semmle.python.frameworks.Stdlib
+
+/**
+ * INTERNAL: Do not use.
+ *
+ * Provides models for the `djangorestframework` PyPI package
+ * (imported as `rest_framework`)
+ *
+ * See
+ * - https://www.django-rest-framework.org/
+ * - https://pypi.org/project/djangorestframework/
+ */
+private module RestFramework {
+  // ---------------------------------------------------------------------------
+  // rest_framework.views.APIView handling
+  // ---------------------------------------------------------------------------
+  /**
+   * An `API::Node` representing the `rest_framework.views.APIView` class or any subclass
+   * that has explicitly been modeled in the CodeQL libraries.
+   */
+  private class ModeledApiViewClasses extends Django::Views::View::ModeledSubclass {
+    ModeledApiViewClasses() {
+      this = API::moduleImport("rest_framework").getMember("views").getMember("APIView") or
```

Comment:
Yes, I was a bit surprised the first time I saw this as well, but apparently there are 2 accepts way of using or:

```
foo()
or
bar()
or
baz()
```

or

```
foo() or
bar() or
baz()
```

Slightly embarrassingly I had problems getting the quick-eval SubclassFinder query to output `\n` without tripping up the extension, so that why I ended up with this way of doing this (which is certainly non-standard for the python CodeQL code)

Reply:
Also note that the autoformatter runs as part of our tests, so it would complain if the formatting was wrong

Reply:
I would prefer that it generated code that looked more familiar, but if it is legal, we can fix the aesthetics later. (Indeed I was confused about the autoformat check not firing..)

Reply:
Ask and you shall receive :blush: